### ISSUE TITLE
[Android] Talkback does not read text characters when navigating between letters or words in a textfield.  

### ISSUE THREAD

**Issue by luomo-pro** [flutter/flutter#93110]
I created a TextField and entered some content in it. Later, I used the character mode of talk back to browse the characters. The cursor moved, but it will not read the characters at the current cursor position.
In the EditText of the native control, it can be read normally.
This is my code snippet:
```
    TextField(
      autofocus: true,
      decoration: InputDecoration(
        labelText: "user name",
        hintText: "user name or email",
        prefixIcon: Icon(Icons.person)
      ),
 ```     
Is there a way to solve it?
This has a great impact on blind users.
I use：
Flutter (Channel beta, 2.7.0-3.1.pre

**Comment by maheshmnj** [comment]
Hi @luomo-pro, Thanks for filing the issue. 
> I used the character mode of talk back to browse the characters. The cursor moved, but it will not read the characters at the current cursor position.

I could not find any such mode (character mode) could you please share a bit if we have to enable something for that feature?

**Comment by luomo-pro** [comment]
hi
@maheshmnj
First of all, thank you for being willing to solve this problem.
Because I can't speak English, I use translation, so the description may not be very accurate, if you have any questions later, please feel free to ask.
If you're using Version 9.1 of Talk Back, you can swipe a finger up or down the screen and it will read different modes, one of which is character mode.
Depending on the language, you might not be saying character mode, but the effect is the same.
When you switch to this mode, in TextField, a finger slides left or right across the screen and it moves the cursor.
If gestures are not available to you, there is an easier way to trigger the same effect by simply pressing volume plus and volume Minus while the focus is on the TextField.
You can contrast TextField with the native editText.
EditText reads the character of the cursor position as it moves around
But TextField doesn't read anything.
This problem has a huge impact on blind people editing incorrect content because they can't accurately figure out where the cursor is.
In addition to character mode, there's word mode, which moves words, and line mode, which moves lines.
By the way, I'm at:：
https://medium.com/flutter-community/a-deep-dive-into-flutters-accessibility-widgets-eb0ef9455bc
See Semantics can monitor "onMoveCursorBackwardByCharacter" event, it should look is talk back character mode a character will be triggered.
Now there is no problem moving the cursor, the only problem is that the characters in the cursor position cannot be read aloud.
If you have any further questions, please feel free to ask me. Thank you.

**Comment by maheshmnj** [comment]
@luomo-pro thanks for the info, I verified in a flutter app horizontal swipe does not move the text cursor but volume keys does move but it read the text in the input. The input text is only read when the focus is brought to textfield.


Flutter app | Native app
:-: | :-:
<video src='https://user-images.githubusercontent.com/31410839/140520513-6f1e56c0-b06c-4a97-8015-51550516b41c.mov' width=180/> | <video src='https://user-images.githubusercontent.com/31410839/140518781-c2488a72-f77c-4347-a274-061f73454f78.mov' width=180/>






_In this video I am pressing volume keys to move the cursor but it does not read the input text but in native apps it does_

https://user-images.githubusercontent.com/31410839/140519952-c4a9f42f-7d2f-42ef-8d9c-4870e305d1a5.mov

<details>
<summary>flutter doctor -v</summary>

```
[✓] Flutter (Channel stable, 2.5.3, on macOS 12.0.1 21A559 darwin-arm, locale en-GB)
    • Flutter version 2.5.3 at /Users/mahesh/Documents/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 18116933e7 (3 weeks ago), 2021-10-15 10:46:35 -0700
    • Engine revision d3ea636dc5
    • Dart version 2.14.4

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/mahesh/Library/Android/sdk
    • Platform android-31, build-tools 31.0.0
    • ANDROID_HOME = /Users/mahesh/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6916264)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 13.1, Build version 13A1030d
    • CocoaPods version 1.10.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6916264)

[✓] IntelliJ IDEA Community Edition (version 2021.2.1)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 60.1.4
    • Dart plugin version 212.5080.8

[✓] VS Code (version 1.61.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.28.0

[✓] Connected device (2 available)
    • macOS (desktop) • macos  • darwin-arm64   • macOS 12.0.1 21A559 darwin-arm
    • Chrome (web)    • chrome • web-javascript • Google Chrome 95.0.4638.69

• No issues found!
```

```
[✓] Flutter (Channel master, 2.6.0-12.0.pre.604, on macOS 12.0.1 21A559 darwin-arm, locale en-GB)
    • Flutter version 2.6.0-12.0.pre.604 at /Users/mahesh/Documents/flutter_master
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 7779ad3c60 (8 hours ago), 2021-11-04 23:05:45 -0700
    • Engine revision 5140e30cec
    • Dart version 2.15.0 (build 2.15.0-268.0.dev)

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/mahesh/Library/Android/sdk
    • Platform android-31, build-tools 31.0.0
    • ANDROID_HOME = /Users/mahesh/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6916264)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 13.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • CocoaPods version 1.10.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6916264)

[✓] IntelliJ IDEA Community Edition (version 2021.2.1)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 60.1.4
    • Dart plugin version 212.5080.8

[✓] VS Code (version 1.61.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.28.0

[✓] Connected device (2 available)
    • macOS (desktop) • macos  • darwin-arm64   • macOS 12.0.1 21A559 darwin-arm
    • Chrome (web)    • chrome • web-javascript • Google Chrome 95.0.4638.69

• No issues found!
```
</details>

**Comment by luomo-pro** [comment]
@maheshmnj
I'm glad you checked him out.
I'm a blind developer myself, so sorry I can't watch the video.
But your description is correct.
Hope this problem can be fixed soon.
This is very important for blind users.
Thank you very much!

**Comment by AlexV525** [comment]
cc @justinmc @LongCatIsLooong

**Comment by justinmc** [comment]
Also CC @chunhtai I think.

**Comment by maheshmnj** [comment]
Fixed in https://github.com/flutter/engine/pull/30199
